examples/src/main/java/io/dapr/examples/invoke/grpc/README.md

```diff
@@ -95,55 +93,59 @@ The `app-id` argument is used to identify this service in Dapr's runtime. The `a
-        // Normally a dapr app would be a web service and not exit main.
-        System.out.println("Done");
+
+    String user = "World";
+    // Access a service running on the local machine on port 50051
+    String target = "localhost:50051";
+
+    ManagedChannel channel = Grpc.newChannelBuilder(target, InsecureChannelCredentials.create())
+        .build();
+
+    try {
+      HelloWorldGrpc.HelloWorldBlockingStub blockingStub = HelloWorldGrpc.newBlockingStub(channel);
+
+      Metadata headers = new Metadata();
+      headers.put(Metadata.Key.of("dapr-app-id", Metadata.ASCII_STRING_MARSHALLER),
+          "hellogrpc");
+      blockingStub = MetadataUtils.attachHeaders(blockingStub, headers);
+
+      logger.info("Will try to greet " + user + " ...");
+      try {
+        HelloRequest request = HelloRequest.newBuilder().setName(user).build();
+        HelloReply response = blockingStub.sayHello(request);
+        logger.info("Greeting: " + response.getMessage());
+      } catch (StatusRuntimeException e) {
+        logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
       }
+    } finally {
+      // To prevent leaking resources like threads and TCP connections
+      // the channel should be shut down when it will no longer be used.
+      channel.shutdownNow().awaitTermination(5, TimeUnit.SECONDS);
     }
   }
 ///...
 }
 ```
 
-First, it creates an instance of `DaprClient` via `DaprClientBuilder`. The protocol used by DaprClient is transparent to the application. The HTTP and GRPC ports used by Dapr's sidecar are automatically chosen and exported as environment variables: `DAPR_HTTP_PORT` and `DAPR_GRPC_PORT`. Dapr's Java SDK references these environment variables when communicating to Dapr's sidecar. The Dapr client is also within a try-with-resource block to properly close the client at the end.
-
-Finally, it will go through in an infinite loop and invoke the `say` method every second. Notice the use of `block()` on the return from `invokeMethod` - it is required to actually make the service invocation via a [Mono](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html) object.
-
 Finally, open a new command line terminal and run the client code to send some messages.
 
 <!-- STEP
 name: Run demo client
 expected_stdout_lines:
-  - '== APP == Sending message: Message #0'
-  - '== APP == Message sent: Message #0'
-  - '== APP == Sending message: Message #1'
-  - '== APP == Message sent: Message #1'
+  - '== APP == INFO: Will try to greet World ...'
+  - '== APP == INFO: Greeting: Hello World'
 background: true
 sleep: 10
 -->
 
 ```bash
-dapr run --app-id invokegrpc -- java -jar target/dapr-java-sdk-examples-exec.jar io.dapr.examples.invoke.grpc.HelloWorldClient
+dapr run --app-id invokegrpc --app-protocol grpc --dapr-grpc-port 50051 -- java -jar target/dapr-java-sdk-examples-exec.jar io.dapr.examples.invoke.grpc.HelloWorldClient

```

Comment:
No need to set the port, CLI picks one automatically.

Reply:
Modified, Thanks.